### DISCUSSION TITLE
QR only

### DISCUSSION THREAD

**Discussion by milanzp** [schoero/SwissQRBill#374] [Ideas]
I'm already using SwissQRBill to create svg and that works like a charm. Now, I have a requirement where I would need QR code only, not the whole bill.

Is there a possibility to implement an option to get the QR code only? Looking at the svg structure, it doesn't look too complicated, it looks like a cheap feature.

For the time being, I can select only QR code svg from whole svg and use it, but it's not such a nice solution.

**Comment by schoero** [comment]
This could potentially already be possible (even though a bit hacky) by calling the `generateQRCode()` function directly.
Note that you have to ensure the correctness of the data manually, but you can use the built in functions `cleanData()` and `validateData()` for that.

Furthermore, you have to create the svg, and the swiss cross overlay manually. You could do this using my library [svg-engine](https://github.com/schoero/svg-engine) or by just hardcoding it as a string.

Take a look at this [jsfiddle](https://jsfiddle.net/sw3nv08L/) where you can see what I mean.

Depending on how you import the library, it may not be already possible, since the mentioned functions are not exported in the [package.json](https://github.com/schoero/SwissQRBill/blob/af5245eab0937dc55698ae8fd0d3fb60ff16b93d/package.json#L9).

What is your use case for this feature? If there is a valid use case I may consider to add a native way into the library.

**Reply by milanzp** [reply]
Thanks for the reply!

I'm using `package.json`...

As I said, I have a workaround, where I get only QR code svg by selecting it from the whole bill. Maybe there are some other libraries that could solve my problem, but I don't want to add another dependency for such a simple thing, so I prefer solving it this hacky way.

My use case shortly:
I only need to share account details - so without setting the debtor and amount.
(edit: And it's mobile app, so wasting screen real estate to show whole bill, even though we need only QR code, in this case is not an option)

With an option to crate whole invoice, get the bill in svg or pdf, this is quite powerful library and solves a lot of use cases. My post was basically just an idea how you can expand the usability a bit more, without investing too much time and effort. I guess I'll not be the only guy who needs both QRBill and only QR code. :)
